src/crush/CrushTester.cc

```diff
@@ -371,10 +371,11 @@ int CrushTester::test_with_crushtool(const char *crushtool_cmd,
     "--min-x", "1",
     "--max-x", "50",
     NULL);
+  string opt_ruleset = stringify(ruleset);
   if (ruleset >= 0) {
     crushtool.add_cmd_args(
       "--ruleset",
-      stringify(ruleset).c_str(),
+      opt_ruleset.c_str(),
```

Comment:
> Temporary objects are destroyed as the last step in evaluating the full-expression (1.9) that (lexically) contains the point where they were created.

see § 12.2/3, n3337 (http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2012/n3337.pdf), and http://en.cppreference.com/w/cpp/language/lifetime